yesod-core/src/Yesod/Core/Internal/TH.hs

```diff
@@ -160,7 +160,7 @@ mkYesodGeneral appCxt' namestr mtys isSub f resS = do
             , renderRouteDec
             , [routeAttrsDec]
             , resourcesDec
-            , if isSub then [] else masterTypeSyns vns site
+            , if isSub then [] else masterTypeSyns (map mkName mtys) site
```

Comment:
It feels like `vns` above is always of length 0? The arity of the data constructor should be the same as the amount of arguments that got parsed from `mkYesodWithParser` (modulo some compliated type/data family stuff). We also need to use the exact same names, because those are placed in all the places the generated type is used. Alternatively we could figure out how to place the generated `t1, .. tn`s in both places, because it isn't being done right now.